clients/drcachesim/tools/invariant_checker.h

```diff
@@ -125,13 +125,20 @@ class invariant_checker_t : public analysis_tool_t {
         memref_t last_branch_ = {};
         memtrace_stream_t *stream = nullptr;
         memref_t prev_entry_ = {};
-        memref_t prev_instr_ = {};
-        std::unique_ptr<instr_autoclean_t> prev_instr_decoded_ = nullptr;
         memref_t prev_xfer_marker_ = {}; // Cleared on seeing an instr.
         memref_t last_xfer_marker_ = {}; // Not cleared: just the prior xfer marker.
         uintptr_t prev_func_id_ = 0;
         addr_t last_retaddr_ = 0;
         uintptr_t trace_version_ = 0;
+        struct instr_info_t {
+            memref_t instr;
+            // Decoding related attributes.
+            bool has_decoded = false;
```

Comment:
Is there a clearer name: not sure what this means, and it sounds ungrammatical.  "was_decoded"??  But we're decoding every instr, aren't we?  Is this supposed to be a "valid" field indicating whether the other fields are valid?